admin-dev/themes/new-theme/js/pages/order/view/order-product-add.js

```diff
@@ -0,0 +1,127 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import Router from '../../../components/router';
+import OrderViewPageMap from '../OrderViewPageMap';
+import {EventEmitter} from '../../../components/event-emitter';
+import OrderViewEventMap from './order-view-event-map';
+
+const $ = window.$;
+
+export default class OrderProductAdd {
+  constructor() {
+    this.router = new Router();
+    this.productAddActionBtn = $(OrderViewPageMap.productAddActionBtn);
+    this.productIdInput = $(OrderViewPageMap.productAddIdInput);
+    this.combinationsBlock = $(OrderViewPageMap.productAddCombinationsBlock);
+    this.combinationsSelect = $(OrderViewPageMap.productAddCombinationsSelect);
+    this.priceTaxIncludedInput = $(OrderViewPageMap.productAddPriceTaxInclInput);
+    this.priceTaxExcludedInput = $(OrderViewPageMap.productAddPriceTaxExclInput);
+    this.taxRateInput = $(OrderViewPageMap.productAddTaxRateInput);
+    this.quantityInput = $(OrderViewPageMap.productAddQuantityInput);
+    this.availableText = $(OrderViewPageMap.productAddAvailableText);
+    this.available = null;
+    this.setupListener();
+    this.product = {};
+  }
+
+  setupListener() {
+    this.combinationsSelect.on('change', (event) => {
+      this.priceTaxExcludedInput.val($(event.currentTarget).find(':selected').attr('data-price-tax-excluded'));
+      this.priceTaxIncludedInput.val($(event.currentTarget).find(':selected').attr('data-price-tax-included'));
+      this.available = $(event.currentTarget).find(':selected').attr('data-stock');
+      this.quantityInput.trigger('change');
+    });
+    this.quantityInput.on('change keyup', (event) => {
+      if (this.available === null) {
+        return;
+      }
+      const quantity = parseInt(event.target.value ? event.target.value : 0, 10);
+      const available = this.available - quantity;
+      this.availableText.text(available);
+      this.availableText.toggleClass('text-danger font-weight-bold', available < 0);
+    });
+    this.productIdInput.on('change', () => {
+      this.productAddActionBtn.removeAttr('disabled');
+    });
+    this.priceTaxIncludedInput.on('change keyup', (event) => {
+      let priceTaxIncl = parseFloat(event.target.value);
+      if (priceTaxIncl < 0 || isNaN(priceTaxIncl)) {
+        priceTaxIncl = 0;
+      }
+      const taxRate = this.taxRateInput.val() / 100 + 1;
+      this.priceTaxExcludedInput.val(ps_round(priceTaxIncl / taxRate, 2));
```

Comment:
Just for my information:

`const taxRate = this.taxRateInput.val() / 100 + **1**;`

Why do we need to add 1 ?

Reply:
It's for the tax calculation, for avoiding to have the price to remove or add.